test/saithrift/tests/saiacl.py

```diff
@@ -1634,3 +1634,453 @@ def runTest(self):
+        lag_member_id1 = sai_thrift_create_lag_member(self.client, lag_id1, port1)
+
+        # Creating RIF
+        rif_id1 = sai_thrift_create_router_interface(self.client, vr_id, SAI_ROUTER_INTERFACE_TYPE_PORT, lag_id1, 0, v4_enabled, v6_enabled, mac1)
+        rif_id2 = sai_thrift_create_router_interface(self.client, vr_id, SAI_ROUTER_INTERFACE_TYPE_VLAN, 0, vlan_oid, v4_enabled, v6_enabled, mac1)
+
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        ip_addr1 = '10.10.10.1'
+        ip_addr_subnet1 = '11.11.11.0'
+        ip_mask = '255.255.255.0'
+        dmac1 = '00:11:22:33:44:55'
+        ip_addr2 = '192.168.0.1'
+        ip_addr_subnet2 = '12.12.12.0'
+        dmac2 = '00:11:22:33:44:56'
+
+        # Creating Neighbor, NHop and Route
+        sai_thrift_create_neighbor(self.client, addr_family, rif_id2, ip_addr2, dmac2)
+        sai_thrift_create_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+
+        nhop1 = sai_thrift_create_nhop(self.client, addr_family, ip_addr2, rif_id2)
+        nhop2 = sai_thrift_create_nhop(self.client, addr_family, ip_addr1, rif_id1)
+
+        sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr_subnet1, ip_mask, nhop1)
+        sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr_subnet2, ip_mask, nhop2)
+
+        sai_thrift_create_fdb(self.client, vlan_oid, dmac2, port2, mac_action)
+
+        # send the test packet(s)
+        pkt = simple_tcp_packet(
+                                pktlen=100,
+                                eth_dst=router_mac,
+                                eth_src=dmac1,
+                                ip_src=ip_addr1,
+                                ip_dst='11.11.11.1',
+                                ip_id=105,
+                                ip_ttl=64)
+        exp_pkt = simple_tcp_packet(
+                                pktlen=100,
+                                eth_dst=dmac2,
+                                eth_src=router_mac,
+                                ip_src=ip_addr1,
+                                ip_dst='11.11.11.1',
+                                ip_id=105,
+                                ip_ttl=63)
+        pkt1 = simple_tcp_packet(
+                                pktlen=100,
+                                eth_dst=router_mac,
+                                eth_src=dmac2,
+                                ip_src=ip_addr2,
+                                ip_dst='12.12.12.1',
+                                ip_id=105,
+                                ip_ttl=64)
+        exp_pkt1 = simple_tcp_packet(
+                                pktlen=100,
+                                eth_dst=dmac1,
+                                eth_src=router_mac,
+                                ip_src=ip_addr2,
+                                ip_dst='12.12.12.1',
+                                ip_id=105,
+                                ip_ttl=63)
+
+
```

Comment:
Please add declaration for object id acl_table_group_id1, acl_table_group_id2 and other objects that will be removed in finally block to eliminate potential 'referenced before assignment' error when test fails.

Please also find pull request #979 as a reference.

Reply:
Addressed review comments